src/NHibernate.Test/NHSpecificTest/GH3325/Mappings.hbm.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<hibernate-mapping xmlns="urn:nhibernate-mapping-2.2" assembly="NHibernate.Test"
+                   namespace="NHibernate.Test.NHSpecificTest.GH3325">
+
+  <class name="Entity">
+    <id name="Id" generator="guid.comb"/>
+    <version name="Version" column="Version" type="Int32" unsaved-value="-1" />
+    <property name="Name"/>
+    <set name="Children" cascade="persist,delete,save-update,evict,lock,replicate,delete-orphan" collection-type="PersistentSetType`1[ChildEntity]">
```

Comment:
Why the custom collection type? Is-it required to trigger the trouble?

Reply:
Bette remove it then, cherry pick the commit on a branch without your fix, and check it triggers the trouble.

Reply:
I've removed the custom collection from the mapping
Problem is independent of the usage of such a collection